### DISCUSSION TITLE
[7.x] Calling getOriginal With Key Parameter That Does Not Exist In Raw Originals Should Throw An Exception

### DISCUSSION THREAD

**Discussion by MannikJ** [laravel/framework#32968] [General]
I really appreciate the new automatic casting when using `getOrignal`, but I think with these changes some unexpected behavior was introduced. 
I know there have been some related issues here and there, but I think they do not exactly cover my point.   

First consider an attribute accessor like the following

```php
public function getFooAttribute()
{
    return "foo";
}
``` 

In previous laravel versions a call like `$model->getOriginal('foo)` would just have returned `null`, because this attributes is not present in the original attributes array.

However in current versions of laravel it will search and find the accessor, call it and return `"foo"`.

One could argue that this expected behavior, because the accessor returns a constant value, so it has obviously not changed in the meantime.

But consider what happens, when you define an accessor like the [full name attribute example](https://laravel.com/docs/7.x/eloquent-mutators#defining-an-accessor) from the docs:

```php
public function getFullNameAttribute()
{
    return "{$this->first_name} {$this->last_name}";
}
```

When I now call `$model->getOriginal('full_name)` I will always get the name created from the current instead of the original state. And as far as I understand the new mechansim does not offer any neat means to solve this problem, because it would only pass the original value to the function if its key matches with the custom attribute name.

So my simple suggestion would be to generally throw an exception if one tries to call `getOriginal` with a key that is not present in the original array to avoid calculating with wrong values without noticing.

**Comment by driesvints** [comment]
Moving this to discussions.
